src/plugins.c

```diff
@@ -2000,7 +2000,7 @@ gboolean geany_plugin_register_proxy(GeanyPlugin *plugin, const gchar **extensio
 	foreach_list(node, active_proxies.head)
 	{
 		proxy = node->data;
-		g_return_if_fail(p != proxy->plugin);
+		g_return_val_if_fail(p != proxy->plugin, FALSE);
```

Comment:
Registering twice is currently explicitly unsupported currently, so any attempt is a programming error (though a plugin error so an assert might not be ideal)

The plan is to enable it once the details are fleshed out and code is written, with the intention to enable proxies to change their list of supported files